File: algorithms/0121. Best Time to Buy and Sell Stock.cpp
class Solution
{
public:
    int maxProfit(vector<int>& prices)
    {
        int ans = 0;
        
        if (prices.size() > 1)
        {
            int min_p = prices[0];
            
            for (int i = 1; i < prices.size(); ++i)
            {
                ans = max(ans, prices[i] - min_p);
                min_p = min(min_p, prices[i]);
            }
        }
        
        return ans;
    }
};

File: 0121. Best Time to Buy and Sell Stock.cs
public class Solution
{
    public int MaxProfit(int[] prices)
    {
        var profit = 0;
        var minPrice = prices[0];

        foreach (var price in prices[1..])
        {
            profit = Math.Max(profit, price - minPrice);
            minPrice = Math.Min(minPrice, price);
        }

        return profit;
    }
}